guides/source/active_record_basics.md

```diff
@@ -228,126 +269,183 @@ CRUD: Reading and Writing Data
+
+The examples below show a few of the CRUD methods as well as the resulting SQL statements.
 
 ### Create
 
 Active Record objects can be created from a hash, a block, or have their
 attributes manually set after creation. The `new` method will return a new
 object while `create` will return the object and save it to the database.
 
-For example, given a model `User` with attributes of `name` and `occupation`,
+For example, given a `Book` model with attributes of `title` and `author`,
 the `create` method call will create and save a new record into the database:
 
 ```ruby
-user = User.create(name: "David", occupation: "Code Artist")
+book = Book.create(title: "The Lord of the Rings", author: "J.R.R. Tolkien")
 ```
 
-Using the `new` method, an object can be instantiated without being saved:
+While the `new` method will instantiate an object without saving it to the database:
 
 ```ruby
-user = User.new
-user.name = "David"
-user.occupation = "Code Artist"
-```
+book = Book.new
+book.title = "The Hobbit"
+book.author = "J.R.R. Tolkien"
 
-A call to `user.save` will commit the record to the database.
+# The above `book` is not yet saved to the database
+
+book.save
+
+# Now the `book` record is committed to the database.
+```
 
 Finally, if a block is provided, both `create` and `new` will yield the new
 object to that block for initialization, while only `create` will persist
 the resulting object to the database:
 
 ```ruby
-user = User.new do |u|
-  u.name = "David"
-  u.occupation = "Code Artist"
+book = Book.new do |b|
+  b.title = "Metaprogramming Ruby 2"
+  b.author = "Paolo Perrotta"
 end
+
+book.save
+```
+
+The resulting SQL statement from both `book.save` and `Book.create` look something like this:
+
+```sql
+INSERT INTO "books" ("title", "author", "created_at", "updated_at") VALUES (?, ?, ?, ?) RETURNING "id"  [["title", "Metaprogramming Ruby 2"], ["author", "Paolo Perrotta"], ["created_at", "2024-02-22 20:01:18.469952"], ["updated_at", "2024-02-22 20:01:18.469952"]]
 ```
 
 ### Read
 
-Active Record provides a rich API for accessing data within a database. Below
-are a few examples of different data access methods provided by Active Record.
+Active Record provides a rich API for accessing data within a database. You can query a single record or multiple records, filter them by any attribute, order them, group them, select specific fields, and do anything you can do with SQL.
 
 ```ruby
-# return a collection with all users
-users = User.all
+# Return a collection with all books
+books = Book.all
 ```
 
 ```ruby
-# return the first user
-user = User.first
+# Return a single book
+first_book = Book.first
+last_book = Book.last
+random_book = Book.take
 ```
 
+The above results in the following SQL:
+
+```sql
```

Comment:
Nit: there's no SQL example for `Book.all`, maybe could combine those two Ruby blocks, and add one line here for the resulting SQL query as well, for completeness?

Reply:
Good point. Added SQL for `Book.all`. I wanted to keep the SQL block separate, as a way to show that the user doesn't need to worry about it to use Active Record. As in "FYI, this is the SQL that's generated behind the scenes."